modules/rostests/apitests/ntdll/RtlGetNtProductType.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for the RtlGetNtProductType API
+ * COPYRIGHT:       Copyright 2020 Bișoc George <fraizeraust99 at gmail dot com>
+ */
+
+#include "precomp.h"
+#include <winbase.h>
+#include <winreg.h>
+
+static
+BOOLEAN
+ReturnNtProduct(PDWORD ProductNtType)
+{
+    LONG Result;
+    HKEY Key;
+    WCHAR Data[20];
+    DWORD Size = sizeof(Data);
+
+    Result = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                           L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions",
+                           0,
+                           KEY_QUERY_VALUE,
+                           &Key);
+    if (Result != ERROR_SUCCESS)
+    {
+        *ProductNtType = 0;
+        return FALSE;
+    }
+
+    Result = RegQueryValueExW(Key,
+                              L"ProductType",
+                              NULL,
+                              NULL,
+                              (PBYTE)&Data,
+                              &Size);
+    if (Result != ERROR_SUCCESS)
+    {
+        RegCloseKey(Key);
+        *ProductNtType = 0;
+        return FALSE;
+    }
+
+    if (wcscmp(Data, L"WinNT") == 0)
+    {
+        *ProductNtType = NtProductWinNt;
+        RegCloseKey(Key);
+        return TRUE;
+    }
+
+    if (wcscmp(Data, L"LanmanNT") == 0)
+    {
+        *ProductNtType = NtProductLanManNt;
+        RegCloseKey(Key);
+        return TRUE;
+    }
+
+    if (wcscmp(Data, L"ServerNT") == 0)
+    {
+        *ProductNtType = NtProductServer;
+        RegCloseKey(Key);
+        return TRUE;
+    }
+
+    *ProductNtType = 0;
+    RegCloseKey(Key);
+    return FALSE;
+}
+
+START_TEST(RtlGetNtProductType)
+{
+    BOOLEAN Status, Ret;
+    DWORD ProductNtType;
+    NT_PRODUCT_TYPE ProductType = NtProductWinNt;
+
+    /*
+     * Wrap the call in SEH. This ensures the testcase won't crash but also
+     * it proves to us that RtlGetNtProductType() throws an exception if a NULL
+     * argument is being passed to caller as output.
+     */
+    StartSeh()
+        RtlGetNtProductType(NULL);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+
+    /* Query the product type normally from the Registry */
+    Ret = ReturnNtProduct(&ProductNtType);
+    if (!Ret)
+    {
+        ok(Ret, "Failed to query the product type value!\n");
+    }
+
+    /* Now, get the product type from the NTDLL system call */
+    Status = RtlGetNtProductType(&ProductType);
```

Comment:
In NT land we tend to reserve the name `Status` for `NTSTATUS` variables.
You could just re-use `Ret` here.